skyportal/handlers/api/annotation.py

```diff
@@ -122,30 +183,62 @@ def post(self):
             )
 
         author = self.associated_user_object
-        annotation = Annotation(
-            data=annotation_data,
-            obj_id=obj_id,
-            origin=origin,
-            author=author,
-            groups=groups,
-        )
+        if associated_resource_type.lower() == "sources":
+            obj_id = resource_id
+            annotation = Annotation(
+                data=annotation_data,
+                obj_id=obj_id,
+                origin=origin,
+                author=author,
+                groups=groups,
+            )
+            annotation_resource_id_str = str(annotation.obj_id)
+        else:
+            return self.error(f'Unknown resource type "{associated_resource_type}".')
```

Comment:
Appropriate HTTP status

Reply:
What's the HTTP status for a wrong resource type? I would think a 400 is good enough?

Reply:
400 here is ok